docs/upgrading.md

```diff
@@ -1,5 +1,7 @@
 # Upgrading
 
+- [Upgrading to v0.103.0-sumo-0](#upgrading-to-v01030-sumo-0)
```

Comment:
since the current version is [102](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.102.0),  i'm guessing it'll be v103 next? and not 102.1 or something 🤔

Reply:
We should merge this PR after the change will be in the upstream. The change may not reach `v0.103.0`

Reply:
Should be `v0.104.0` 😅

Reply:
😆 whoops